tetris/index.js

```diff
@@ -0,0 +1,121 @@
+var express = require('express');
+var app = express();
+var http = require('http').Server(app);
+var io = require('socket.io')(http);
+var path = require('path');
+var util = require('util');
+
+var users = require(__dirname + '/js/userContainer.js')();
+var players = require(__dirname + '/js/playerContainer.js')();
+
+app.use("/scripts", express.static(__dirname + '/public/js'));
+app.use("/styles", express.static(__dirname + '/public/css'));
+
+app.get('/', function(req, res) {
+    res.sendFile('index.html', { root: __dirname });
+});
+
+io.on('connection', function(socket) {
+	
+	socket.on('disconnect', function() {
+		updatePlayers();
+		updateClientUserList();
+		
+		if (users.allUsers[socket.id].showDisconnectMessage)
+			io.emit('message', users.nameById(socket.id) + ' Has Disconnected');	
+	});
+	
+	socket.on('authenticate', function(token) {
```

Comment:
Extract a named function instead of using an anonymous function.

`socket.on('authenticate', authenticate(token))`
Lines should do one thing, and one thing only.
This line both defines the behaviour of authentication, and registers the authenticate function.
More details here: http://callbackhell.com/